cmgd/cmgd_bcknd_adapter.h

```diff
@@ -25,24 +25,113 @@
 #include "cmgd/cmgd_defines.h"
 #include "lib/cmgd_bcknd_client.h"
 
+#define CMGD_FIND_ADAPTER_BY_INDEX(adapter_index)	\
+	cmgd_adaptr_ref[adapter_index]
+
+typedef enum cmgd_bcknd_client_index_ {
+	CMGD_CLIENT_INDEX_ZEBRAD,
+	CMGD_CLIENT_INDEX_VRRPD,
+	CMGD_CLIENT_INDEX_STATICD,
+	CMGD_CLIENT_INDEX_SHARPD,
+	CMGD_CLIENT_INDEX_RIPNGD,
+	CMGD_CLIENT_INDEX_RIPD,
+	CMGD_CLIENT_INDEX_PIMD,
+	CMGD_CLIENT_INDEX_PBRD,
+	CMGD_CLIENT_INDEX_PATHD,
+	CMGD_CLIENT_INDEX_OSPFD,
+	CMGD_CLIENT_INDEX_OSPF6D,
+	CMGD_CLIENT_INDEX_NHRPD,
+	CMGD_CLIENT_INDEX_LDPD,
+	CMGD_CLIENT_INDEX_ISISD,
+	CMGD_CLIENT_INDEX_FABRICD,
+	CMGD_CLIENT_INDEX_EIGRPD,
+	CMGD_CLIENT_INDEX_BGPD,
+	CMGD_CLIENT_INDEX_BABELD,
+	CMGD_CLIENT_INDEX_BFDD,
+	CMGD_CLIENT_INDEX_MAX
+} cmgd_bcknd_client_index_t;
+
+// List of adapter clients of CMGD
+#define CMGD_CLIENT_ID_ZEBRAD (1 << CMGD_CLIENT_INDEX_ZEBRAD)
+#define CMGD_CLIENT_ID_VRRPD (1 << CMGD_CLIENT_INDEX_VRRPD)
+#define CMGD_CLIENT_ID_STATICD (1 << CMGD_CLIENT_INDEX_STATICD)
+#define CMGD_CLIENT_ID_SHARPD (1 << CMGD_CLIENT_INDEX_SHARPD)
+#define CMGD_CLIENT_ID_RIPNGD (1 << CMGD_CLIENT_INDEX_RIPNGD)
+#define CMGD_CLIENT_ID_RIPD (1 << CMGD_CLIENT_INDEX_RIPD)
+#define CMGD_CLIENT_ID_PIMD (1 << CMGD_CLIENT_INDEX_PIMD)
+#define CMGD_CLIENT_ID_PBRD (1 << CMGD_CLIENT_INDEX_PBRD)
+#define CMGD_CLIENT_ID_PATHD (1 << CMGD_CLIENT_INDEX_PATHD)
+#define CMGD_CLIENT_ID_OSPFD (1 << CMGD_CLIENT_INDEX_OSPFD)
+#define CMGD_CLIENT_ID_OSPF6D (1 << CMGD_CLIENT_INDEX_OSPF6D)
+#define CMGD_CLIENT_ID_NHRPD (1 << CMGD_CLIENT_INDEX_NHRPD)
+#define CMGD_CLIENT_ID_LDPD (1 << CMGD_CLIENT_INDEX_LDPD)
+#define CMGD_CLIENT_ID_ISISD (1 << CMGD_CLIENT_INDEX_ISISD)
+#define CMGD_CLIENT_ID_FABRICD (1 << CMGD_CLIENT_INDEX_FABRICD)
+#define CMGD_CLIENT_ID_EIGRPD (1 << CMGD_CLIENT_INDEX_EIGRPD)
+#define CMGD_CLIENT_ID_BGPD (1 << CMGD_CLIENT_INDEX_BGPD)
+#define CMGD_CLIENT_ID_BABELD (1 << CMGD_CLIENT_INDEX_BABELD)
+#define CMGD_CLIENT_ID_BFDD (1 << CMGD_CLIENT_INDEX_BFDD)
+
+// XPATH regular expression map index
+typedef enum cmgd_bcknd_xpath_regexp_id_ {
```

Comment:
Why do we need regexp id?

Reply:
As discussed its used for readability while comparing the expression and easy to understand the derivation process.